docs/development.md

```diff
@@ -52,6 +52,138 @@ git push -f origin myfeature
+				Kind: "Service",
+				Name: serviceName,
+			},
+		},
+	}
+	r, err := c.routeClient.Routes(c.namespace).Create(route)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating route")
+	}
+	return r, nil
+}
+
+```
+Ref: https://github.com/redhat-developer/odo/blob/v0.0.8/pkg/occlient/occlient.go#L1213,L1233
+
+Let's look at the function body and identify how many API calls it is making while execution. 
+In this case CreateRoute is making only a single API call which is 
+` r, err := c.routeClient.Routes(c.namespace).Create(route)`
+
+first we have to add the code for initialising fakeclientset & fakeclient for routeclient on the FakeNew function 
+
+for example,
+```
+            fkclientset.RouteClientset = fakeRouteClientset.NewSimpleClientset()
+            client.routeClient = fkclientset.RouteClientset.Route()
+```
+Ref: https://github.com/redhat-developer/odo/pull/456/files#diff-c1467d30f892daf34a2d0a5641bd41c7R20
+
+initialising a fakeclientset by calling fakeRouteClientset.NewSimpleClientset() 
+we get as simple set of object tracker which can process creates,updates and deletions 
+but without any validations. So it's always better to implement validations separately if needed.
+
+Okey, now we have all we needed for writing unit test function for CreateRoute,
+Let's look at unit test function now
+```
+fkclient, fkclientset := FakeNew()
+```
+Ref: https://github.com/redhat-developer/odo/pull/456/files#diff-54c1e3725d2cfb565cbd1cfdb02bd792R46
+
+The fakeclientset and fakeclient is being initialised 
+and we can make the method calls on the struct which get initialised using FakeNew().
+Here in this case we need to validate the value with which the action is performed 
+and the number of actions performed on routeclientset
+
+We can achieve that by calling the function we want to test on fakeclient which we initialized on the recent step.
+
+Then after making that function call, the values with which its been called is 
+being validated here in this step 
+
+```
+createdRoute := fkclientset.RouteClientset.Actions()[0].(ktesting.CreateAction).GetObject().(*routev1.Route)
+```
+Ref: https://github.com/redhat-developer/odo/pull/456/files#diff-54c1e3725d2cfb565cbd1cfdb02bd792R62
+
+
+For the API calls which are returning objects that are later being processed inside the function body,
+adding reactors for the relevent actions is the way to go with this situation.
+
+for example take a look at `RemoveVolumeFromDeploymentConfig` 
+https://github.com/redhat-developer/odo/blob/master/pkg/occlient/occlient.go#L1413
+
+here its calling GetDeploymentConfigFromName for fetching dc from the server
+which is returning a dc object which is later processed in the function.
+
+```
+func (c *Client) GetDeploymentConfigFromName(name string) (*appsv1.DeploymentConfig, error) {
+    return c.appsClient.DeploymentConfigs(c.namespace).Get(name, metav1.GetOptions{})
+    }
+```
+
+so for this we can add a reactor like below, 
```

Comment:
I think instead of using `RemoveVolumeFromDeploymentConfig` which internally calls `GetDeploymentConfigFromName`, we should have examples of functions which directly make `create`, `list` and `get` calls.